tests/versioning_test.go

```diff
@@ -1535,16 +1564,33 @@ func (s *versioningIntegSuite) TestWorkflowGetsStuckWhenDisablingLoadingUserData
 
 	run, err := s.sdkClient.ExecuteWorkflow(ctx, sdkclient.StartWorkflowOptions{
 		TaskQueue:                tq,
-		WorkflowExecutionTimeout: 5 * time.Second,
+		WorkflowExecutionTimeout: 10 * time.Second,
 	}, "wf")
 	s.Require().NoError(err)
-	err = run.Get(ctx, nil)
-	var timeoutError *temporal.TimeoutError
-	s.Require().ErrorAs(err, &timeoutError)
+
+	// should not run on versioned worker
+	time.Sleep(2 * time.Second)
```

Comment:
I have a couple of issues here:

1. Relying on sleeps in tests is a recipe for flakiness, I'd rather avoid that if possible.
2. More importantly, should we treat versioned workers as unversioned when user data is disabled? If we ever turn on the kill switch, it's going to be very painful for users, requiring new worker deployments.

Reply:
1. I mean, sure, ideally. But we're pretty far gone down the timing road in these tests. Do you have a specific suggestion for avoiding sleeps here? Note that the existing test also implicitly relied on a sleep: the 5s timeout.
2. Well, the current version of the PR does not send versioned tasks to unversioned anymore. So it's still going to be painful but in a different way (refreshing tasks). I'm open to ideas here but we discussed them all before and I'm not sure we're going to come to a different conclusion

Reply:
I've been playing with the DLQ idea for versioned tasks and I think it's a better solution. I'll do it in a follow-up PR. This one should be merged first since it fixes an existing bug and adds more tests.

Reply:
> Note that the existing test also implicitly relied on a sleep: the 5s timeout.

That's not a flaky condition though, it should deterministically fail.
I don't have good idea though for an integration test that can't inspect the internal state of the system.

For (2), let's keep the decision not to break the semantics.

Reply:
The timeout is just as flaky as this 2s sleep: both of them can spuriously pass, but not spuriously fail. It's just the history service noticing that 5s has passed without the workflow completing yet, vs this test noticing that 2s has passed without the workflow running yet.